schulcloud/Controllers/Dashboard/DashboardViewController.swift

```diff
@@ -179,16 +161,17 @@ extension DashboardViewController {
         } else {
             cell.contentView.layer.cornerRadius = 0.0
         }
+        cell.layoutSubviews()
         return cell
     }
 
     override func collectionView(_ collectionView: UICollectionView, didSelectItemAt indexPath: IndexPath) {
-        let vc = viewControllers[indexPath.row]
-        if vc == calendarOverview {
+        let vc = (viewControllers[indexPath.row])
+        if let _ = vc as? PermissionManagmentViewController<CalendarOverviewViewController> {
```

Comment:
please use `is` and a switch statement

```swift
switch vc.containedViewController {
case is HomeworkOverviewViewController:
...
```

for this to work you have to change the type of `self.viewControllers` to 
```swift
var viewControllers : [PermissionAbleViewController] = []
```

Reply:
`PermissionManagmentViewController` does not conform to `PermissionAbleViewController`, there for your approach won't work. My current solution is to use a switch statement the way you describe though.

Reply:
Oh I think I mixed something up then